packages/react-form/src/hooks/test/form-with-dynamic-list.test.tsx

```diff
@@ -142,4 +143,111 @@ describe('useForm with dynamic list', () => {
+
+      wrapper.find('button', {children: 'Add item'}).trigger('onClick');
+      fillRequiredFields(wrapper);
+
+      expect(isDirty(wrapper)).toBe(true);
+
+      await waitForSubmit(wrapper, promise);
+
+      expect(isDirty(wrapper)).toBe(true);
+    });
+
+    it('sets dirty to false after successful submit if makeCleanAfterSubmit is true', async () => {
+      const promise = Promise.resolve(submitSuccess());
+      const wrapper = mount(
+        <FormWithDynamicVariantList
+          data={fakeProduct()}
+          makeCleanAfterSubmit
+        />,
+      );
+
+      wrapper.find('button', {children: 'Add item'}).trigger('onClick');
+      fillRequiredFields(wrapper);
+
+      expect(isDirty(wrapper)).toBe(true);
+
+      await wrapper
+        .find('button', {type: 'submit'})!
+        .trigger('onClick', clickEvent());
+      expect(isDirty(wrapper)).toBe(false);
+    });
+
+    it('does not set dirty to false after successful submit if makeCleanAfterSubmit is false', async () => {
+      const promise = Promise.resolve(submitSuccess());
+      const wrapper = mount(
+        <FormWithDynamicVariantList
+          data={fakeProduct()}
+          makeCleanAfterSubmit={false}
+        />,
+      );
+
+      wrapper.find('button', {children: 'Add item'}).trigger('onClick');
+      fillRequiredFields(wrapper);
+
+      expect(isDirty(wrapper)).toBe(true);
+
+      await waitForSubmit(wrapper, promise);
+
+      expect(isDirty(wrapper)).toBe(true);
+    });
+
+    it('does not set dirty to false if makeCleanAfterSubmit is true but submit is unsuccessful', async () => {
+      const promise = Promise.resolve(submitFail());
+      const wrapper = mount(
+        <FormWithDynamicVariantList
+          data={fakeProduct()}
+          makeCleanAfterSubmit={false}
+        />,
+      );
+
+      wrapper.find('button', {children: 'Add item'}).trigger('onClick');
+      fillRequiredFields(wrapper);
+
+      expect(isDirty(wrapper)).toBe(true);
+
+      await waitForSubmit(wrapper, promise);
+
+      expect(isDirty(wrapper)).toBe(true);
+    });
+  });
+
+  describe('makeClean', () => {
+    it('cleans the forms dirty state', () => {
+      const wrapper = mount(
+        <FormWithDynamicVariantList data={fakeProduct()} />,
+      );
+
+      wrapper.find('button', {children: 'Add item'}).trigger('onClick');
+      fillRequiredFields(wrapper);
+
+      expect(isDirty(wrapper)).toBe(true);
+
+      wrapper
+        .find('button', {children: 'Clean'})!
+        .trigger('onClick', clickEvent());
+
+      expect(isDirty(wrapper)).toBe(false);
+    });
+  });
 });
+
+function fillRequiredFields(wrapper: Root<any>) {
+  const optionTextFields = wrapper.findAll(TextField, {label: 'option'});
+  optionTextFields.forEach(textField =>
+    textField.trigger('onChange', 'a value'),
+  );
+
+  const titleTextFields = wrapper.findAll(TextField, {label: 'title'});
+  titleTextFields.forEach(textField =>
+    textField.trigger('onChange', 'a value'),
+  );
+}
```

Comment:
Please move this to utilities maybe? I would rename to `fillAllOptionAndTitleFields`

Reply:
Moved it but I am thinking we should keep the name `fillRequiredFields` just in case more required fields are added.